code/places-monitor-android/src/phone/java/com/adobe/marketing/mobile/PlacesActivity.java

```diff
@@ -34,10 +35,68 @@
 
 public class PlacesActivity extends Activity {
 
+
 	private static final String FINE_LOCATION = Manifest.permission.ACCESS_FINE_LOCATION;
 	private static final String BACKGROUND_LOCATION = Manifest.permission.ACCESS_BACKGROUND_LOCATION;
 	private static final String INTENT_PERMISSION_KEY = "intent permission key";
 
+
+	/**
+	 * Update the location authorization status held by the Places extension.
+	 *
+	 * The status will be stored in the Places shared state and is for reference only.
+	 * Calling this method has no impact on the location authorization status for this device.
+	 */
+	public static void updateLocationAuthorizationStatus () {
+		// for version below API 23, location permission are granted at install time
+		if (!isRuntimePermissionRequired()) {
+			Places.setAuthorizationStatus(PlacesAuthorizationStatus.ALWAYS);
+			return;
+		}
+
+		// get hold of the app context. bail out with unknown status if context is unavailable
+		Context context = App.getAppContext();
+
+		if (context == null) {
+			Log.warning(PlacesMonitorConstants.LOG_TAG, "Unable to check location permission, App context is not available. Defaulting acquired permission level to unknown");
+			Places.setAuthorizationStatus(PlacesAuthorizationStatus.UNKNOWN);
+			return;
+		}
+
+		int permissionStatus = ActivityCompat.checkSelfPermission(context, FINE_LOCATION);
+		if(permissionStatus == PackageManager.PERMISSION_GRANTED) {
+			// if the permission for accessing fine location is granted. one of the following is true
+			// 1. permission is granted to access location only `when app in use` (for API 29 and above)
+			// 2. permission is granted to access location in background
+
+			// for android version below API 29, background location permission are granted by default
+			if(isBackgroundLocationAccessGrantedByDefault()){
+				Places.setAuthorizationStatus(PlacesAuthorizationStatus.ALWAYS);
+				return;
+			}
+
+			// for android version above API 29. verify if the access to background location is granted specifically
+			if(isBackgroundPermissionGranted()) {
+				Places.setAuthorizationStatus(PlacesAuthorizationStatus.ALWAYS);
+				return;
+			}
+			Places.setAuthorizationStatus(PlacesAuthorizationStatus.WHEN_IN_USE);
+			return;
```

Comment:
the comments here are great for explaining decisions being made!  let's simplify the code:

    if (isBackgroundLocationAccessGrantedByDefault() || isBackgroundPermissionGranted()) {
        Places.setAuthorizationStatus(PlacesAuthorizationStatus.ALWAYS);
    } else {
        Places.setAuthorizationStatus(PlacesAuthorizationStatus.WHEN_IN_USE);
    }

Reply:
thanks for the suggestion.. I cleaned this up and removed a bunch of unwanted `return`